src/Core/Managed/Shared/DataContracts/AvailabilityTelemetry.cs

```diff
@@ -0,0 +1,170 @@
+﻿namespace Microsoft.ApplicationInsights.DataContracts
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.External;
+    using System.Globalization;
+
+    /// <summary>
+    /// Telemetry type used for availability test results.
+    /// Contains a time and message and optionally some additional metadata.
+    /// </summary>
+    public sealed class AvailabilityTelemetry : ITelemetry, ISupportProperties //TODO: Should this implement IOperation like RequestTelemetry?
+    {
+        internal const string TelemetryName = "Availability";
+
+        internal readonly string BaseType = typeof(AvailabilityData).Name;
+        internal readonly AvailabilityData Data;
+        private readonly TelemetryContext context;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvailabilityTelemetry"/> class with empty properties.
+        /// </summary>
+        public AvailabilityTelemetry()
+        {
+            this.Data = new AvailabilityData();
+            this.context = new TelemetryContext(this.Data.properties);
+            this.Data.testRunId = Guid.NewGuid().ToString();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvailabilityTelemetry"/> class with empty properties.
+        /// </summary>
+        public AvailabilityTelemetry(string testName, DateTimeOffset testTimeStamp, TimeSpan duration, string runLocation, bool success, string message = null)
+        {
+            this.Data = new AvailabilityData();
+            this.context = new TelemetryContext(this.Data.properties);            
+            this.Data.testTimeStamp = testTimeStamp.ToString("o", CultureInfo.InvariantCulture);
+            this.Data.testName = testName;
+            this.Data.testRunId = Guid.NewGuid().ToString();
+            this.Data.duration = duration.ToString(string.Empty, CultureInfo.InvariantCulture);
+            this.Data.result = success ? TestResult.Pass : TestResult.Fail;
+            this.Data.runLocation = runLocation;
+            this.Data.message = message;           
+        }        
+
+        /// <summary>
+        /// Gets or sets the test run id guid
+        /// </summary>
+        public Guid Id
+        {
+            get { return Guid.Parse(this.Data.testRunId); }
```

Comment:
>Parse [](start = 30, length = 5)

CoreSDK never throws exceptions. It substitutes it with default values. E.g. in ip address we check that it is ip in v4 format.